streams/src/main/java/org/apache/kafka/streams/kstream/ValueJoiner.java

```diff
@@ -18,13 +18,20 @@
 package org.apache.kafka.streams.kstream;
 
 /**
- * The ValueJoiner interface for joining two values and return a the joined new value.
+ * The {@link ValueJoiner} interface for joining two values into a new value.
```

Comment:
What about:

> Defines a stateless transformation of two record values into a new value.

(Not sure whether this wording approach is consistent with other interface docstrings.)